tests/Unit/Evolution/Ringdown/Test_StrahlkorperCoefsInRingdownDistortedFrame.cpp

```diff
@@ -0,0 +1,182 @@
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+// [[TimeOut, 10]]
+SPECTRE_TEST_CASE("Unit.Evolution.Ringdown.SCoefsRDis", "[Unit][Evolution]") {
+  // Write a temporary H5 file with Strahlkorpers at different times, then
+  // pass that file's path to strahlkorper_coefs_in_ringdown_distorted_frame().
+  // First, if the temporary file exists, remove it
+  const std::string horizons_file_name{"Unit.Evolution.Ringdown.SCoefsRDis.h5"};
+  const std::string horizons_subfile_name{"/ObservationAhC__Ylm.dat"};
+  if (file_system::check_if_file_exists(horizons_file_name)) {
+    file_system::rm(horizons_file_name, true);
+  }
+
+  MAKE_GENERATOR(generator);
+
+  // Start out with a Strahlkorper at rest in a grid frame, then map it
+  // to the inertial frame. The shape map is an identity map (it's initialized
+  // with Schwarzschild coefficients), so the grid->distorted map
+  // here is the identity. But start in a grid frame to use the available
+  // grid->distorted instantiation of strahlkorper_in_different_frame.
+  constexpr const std::array<double, 3> expected_center{4.4, 5.5, 6.6};
+  constexpr const size_t l_max = 12;
+  constexpr const size_t m_max = 12;
+  const auto kerr_horizon_radius = get(gr::Solutions::kerr_horizon_radius(
+      ::ylm::Spherepack(l_max, m_max).theta_phi_points(), 1.0,
+      {{0.0, 0.0, 0.8}}));
+  const auto expected_strahlkorper = ylm::Strahlkorper<Frame::Grid>(
+      l_max, m_max, kerr_horizon_radius, expected_center);
+
+  // Make a set of times to evaluate the functions of time at
+  static constexpr size_t number_of_times{9};
+  const std::array<double, number_of_times> times{0.0, 0.5, 1.0, 1.5, 2.0,
+                                                  2.5, 3.0, 3.5, 4.0};
+
+  // Set match time to earliest time; must be earliest time, since
+  // the grid->inertial map used below will not be valid at times earlier
+  // than the match time
+  const double match_time{0.0};
+
+  // Next, set up a temporary domain (just to hold functions of time)
+  // and some functions of time defining the ringdown Distorted->Inertial map.
+  std::uniform_real_distribution<double> fot_dist{0.1, 0.5};
+  const auto exp_func_and_2_derivs =
+      make_with_random_values<std::array<double, 3>>(make_not_null(&generator),
+                                                     make_not_null(&fot_dist));
+  const auto exp_outer_bdry_func_and_2_derivs =
+      make_with_random_values<std::array<double, 3>>(make_not_null(&generator),
+                                                     make_not_null(&fot_dist));
+  auto initial_unit_quaternion = make_with_random_values<std::array<double, 4>>(
+      make_not_null(&generator), make_not_null(&fot_dist));
+  const double initial_unit_quaternion_magnitude = sqrt(
```

Comment:
[optional] You could use a `boost::math::quaternion<double>` here and then use the helpers in `Domain/FunctionsOfTime/QuaternionHelpers.hpp` to normalize

Reply:
Good to know for next time, but since it's optional, I'd rather leave as is